src/main/java/com/incognia/IncogniaAPI.java

```diff
@@ -28,14 +25,24 @@ public IncogniaAPI(String clientId, String clientSecret, Region region) {
     Asserts.assertNotEmpty(clientId, "client id");
     Asserts.assertNotEmpty(clientSecret, "client secret");
     Asserts.assertNotEmpty(apiUrl, "api url");
-    this.clientId = clientId;
-    this.clientSecret = clientSecret;
-    this.apiUrl = apiUrl;
     // TODO (rato): set client timeout
-    this.httpClient = new OkHttpClient.Builder().build();
+    tokenAwareNetworkingClient =
+        new TokenAwareNetworkingClient(new Builder().build(), apiUrl, clientId, clientSecret);
   }
 
-  // POST onboarding
+  public SignupResponse registerSignup(String installationId, Address address)
```

Comment:
Should we abstract out the idea of response the same way we did for the request? Or the opposite, let the user pass the request object.

Reply:
we probably shouldn't abstract the response, as the pojos would be equal (or very similar). We can evolve this in the future, tough

Reply:
I meant it regarding naming, not necessarily needing to have a `*Response` being mapped to something else.

I'm a little uncomfortable about having a `*Response` without a `*Request`. As it makes sense to not require the user to build a request object, I was wondering if we could return something that isn't called a response. In the proposal for Ruby, this object is called `SignupAssesment`, what do you think of using the same name?

Reply:
Actually, in Ruby the idea is to return `Incognia::Response` which contains `status` and `resource` (the `SignupAssesment` in this case), but for Java, the status will be implicit in case of success or an exception in case of error, right?

Reply:
yeah, nice catch. changing to `SignupAssessment`